python/semantic_kernel/connectors/memory/mongodb_atlas_vector_search/atlas_vector_search_memory_store.py

```diff
@@ -0,0 +1,385 @@
+
+        Returns:
+            List[str] -- A group of collection names.
+        """
+        return await self.database.list_collection_names()
+
+    async def delete_collection_async(self, collection_name: str) -> None:
+        """Deletes a collection from the data store.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+
+        Returns:
+            None
+        """
+        await self.database[collection_name].drop()
+
+    async def does_collection_exist_async(self, collection_name: str) -> bool:
+        """Determines if a collection exists in the data store.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+
+        Returns:
+            bool -- True if given collection exists, False if not.
+        """
+        return collection_name in (await self.get_collections_async())
+
+    async def upsert_async(self, collection_name: str, record: MemoryRecord) -> str:
+        """Upserts a memory record into the data store. Does not guarantee that the collection exists.
+            If the record already exists, it will be updated.
+            If the record does not exist, it will be created.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+            record {MemoryRecord} -- The memory record to upsert.
+
+        Returns:
+            str -- The unique identifier for the memory record.
+        """
+        await self.create_collection_async(collection_name)
+
+        if not record._id:
+            record._id = str(uuid.uuid4())
+
+        document: Mapping[str, Any] = memory_record_to_mongo_document(record)
+
+        update_result: results.UpdateResult = await self.database[
+            collection_name
+        ].update_one(document, {"$set": document}, upsert=True)
+
+        assert update_result.acknowledged
+        return record._id
+
+    async def upsert_batch_async(
+        self, collection_name: str, records: List[MemoryRecord]
+    ) -> List[str]:
+        """Upserts a group of memory records into the data store. Does not guarantee that the collection exists.
+            If the record already exists, it will be updated.
+            If the record does not exist, it will be created.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+            records {MemoryRecord} -- The memory records to upsert.
+
+        Returns:
+            List[str] -- The unique identifiers for the memory records.
+        """
+        await self.create_collection_async(collection_name)
+
+        upserts: List[UpdateOne] = []
+        for record in records:
+            if not record._id:
+                record._id = str(uuid.uuid4())
+                self._logger.debug(
+                    "record id not found for Memory Store, mutated with new uuid {}".format(
```

Comment:
The Python [docs](https://docs.python.org/3/library/logging.html#logging.debug) recommend passing the formatting args as `*args`.

Reply:
Yeah, agreed. However, it looks like the `NullLogger` used here doesn't marry well with that approach. 
```
>       self._logger.debug(
            "matched_count=%s, upserted_count=%s",
            bulk_update_result.matched_count,
            bulk_update_result.upserted_count,
        )
E       TypeError: debug() takes 2 positional arguments but 4 were given
```

I went ahead and looked at the [`NullLogger`](https://github.com/microsoft/semantic-kernel/blob/main/python/semantic_kernel/utils/null_logger.py) implementation and now think it makes sense to have a small PR to change the function signatures of the `NullLogger` to just match the base `Logger` class. 

cc: @awharrison-28 @shawncal for thoughts?

Reply:
That makes sense. @Jibola can you create an issue to track this work?